### PR TITLE
[ENH] pad t and F 'short' contrasts with zeros

### LINKED ISSUE #2396: Length of contrast vectors depends on high cutoff frequency
The length of the contrast vectors in [this example](https://github.com/nistats/nistats/blob/61a86db07429739e0c8e7ee0110dbc7369fe7aa3/examples/01_tutorials/plot_single_subject_single_run.py), lines 156 and 157,

https://github.com/nistats/nistats/blob/61a86db07429739e0c8e7ee0110dbc7369fe7aa3/examples/01_tutorials/plot_single_subject_single_run.py#L156
https://github.com/nistats/nistats/blob/61a86db07429739e0c8e7ee0110dbc7369fe7aa3/examples/01_tutorials/plot_single_subject_single_run.py#L157

depends on the high cutoff frequency set by the user:
https://github.com/nistats/nistats/blob/61a86db07429739e0c8e7ee0110dbc7369fe7aa3/examples/01_tutorials/plot_single_subject_single_run.py#L99

Length of the contrast vectors should only depend on the number of regressors-of-interest. Regressors-of-no-interest should be introduced automatically, according to the parameters provided.

This issue is in reference to the PR nilearn/nistats#351

### PR DESCRIPTION
**Pull request by Remi-Gau**
- Closes #2396

Changes proposed in this pull request:

- pad t and F constrast when they are too short
- throw a warning when padding is done
- throw error only when contrast are longer than the number of columns

### CODE REVIEW

**Comment by Remi-Gau** [comment]
mentioned in the associated issue, using the contrasts symbolically (eg. 'c1 - c2') in the example mentioned in the issue so avoid having to change contrast length when the high pass filter model is changed.

**Comment by Remi-Gau** [comment]
@bthirion I would not mind an early review on this.

I do not like that the warnings / errors that have to be thrown in different places, feels like there should be a single function checking contrast shape and doing the padding rather than changes in several places.

**Comment by bthirion** [comment]
You mean, one single function for t and F ?

**Comment by bthirion** [comment]
The PR looks good overall.

File: nilearn/glm/contrasts.py [hunk 1]
```diff
@@ -106,6 +107,9 @@ def compute_contrast(labels, regression_result, con_val, contrast_type=None):
         for label_ in regression_result:
             label_mask = labels == label_
             reg = regression_result[label_]
+            con_val = pad_contrast(
+                con_val=con_val, theta=reg.theta, contrast_type=contrast_type
+            )
```
Comment (Remi-Gau) [inline nilearn/glm/contrasts.py:112]: @bthirion 
any reason why for F contrasts we do not rely on the FContrast methods and do something like 

```python
reg = regression_result[label_].Fcontrast(con_val)
```

the way we do it for T contrasts above?

Comment (bthirion) [reply nilearn/glm/contrasts.py:112]: IIRC, this is because we had to trick it somehow to support fixed effects on F contrasts.

Comment (Remi-Gau) [reply nilearn/glm/contrasts.py:112]: ok this may require a comment if we can remember the why otherwise I could add a TODO to try to refactor this

Comment (bthirion) [reply nilearn/glm/contrasts.py:112]: I can add this to your PR.

**Comment by Remi-Gau** [comment]
# TODO

- [ ] update changelog

File: nilearn/glm/tests/test_second_level.py [hunk 2]
```diff
@@ -1083,11 +1078,23 @@ def test_second_level_contrast_computation_errors(tmp_path, rng):
...
+    X = pd.DataFrame([[1]] * 4, columns=["intercept"])
+    model = model.fit(Y, design_matrix=X)
+
     with pytest.raises(
         ValueError,
-        match=("t contrasts should be length P=2, but this is length 1"),
+        match=("t contrasts should be of length P=1, but it has length 2."),
     ):
-        model.compute_contrast([1])
+        model.compute_contrast(second_level_contrast=[1, 2])
```
[the remark this answers is not in the document]
Comment (Remi-Gau) [reply nilearn/glm/tests/test_second_level.py:1097]: yup

File: nilearn/glm/contrasts.py (hunk 1 above)
Comment (Remi-Gau) [reply nilearn/glm/contrasts.py:116]: Added a TODO for now in case we want to merge and tackle this in a separate PR.

File: nilearn/glm/_utils.py [hunk 3]
```diff
@@ -343,3 +343,59 @@ def positive_reciprocal(X):
...
+    If the contrast is longer than the number of parameters,
+    a ValueError is raised.
+
+    Parameters
+    ----------
+    con_val : numpy.ndarray of shape (p) or (q, p)
+        Where q = number of :term:`contrast` vectors
+        and p = number of regressors.
+
+    theta theta of RegressionResults instances
```
Comment (ymzayek) [inline nilearn/glm/_utils.py:363]: can you add the type for theta here?

**Review by ymzayek — COMMENTED** [review]
LGTM overall!

File: nilearn/glm/_utils.py (hunk 3 above)
Comment (Remi-Gau) [reply nilearn/glm/_utils.py]: added the type but also tried to improve the doc string: let me know if this reads sufficiently well.

File: nilearn/glm/contrasts.py (hunk 1 above)
Comment (bthirion) [reply nilearn/glm/contrasts.py:116]: Maybe rather in another PR.

**Review by bthirion — APPROVED** [review]
LGTM, thx.
